### PR TITLE
Support same service instance preference

### PR DESCRIPTION
**Pull request by OlgaMaciaszek**
Fixes gh-689.

### CODE REVIEW

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 1]
```diff
@@ -995,6 +995,28 @@ public class CustomLoadBalancerConfiguration {
...
+			ConfigurableApplicationContext context) {
+		return ServiceInstanceListSupplier.builder()
+					.withDiscoveryClient()
+					.withSameInstancePreference()
+					.build(context);
+	    }
+	}
+----
+
+TIP: This is also a replacement for Zookeeper StickyRule.
```
Comment (ryanjbaxter) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:1019]: Probably good to add something to the Zookeeper docs as well

**Review by ryanjbaxter — APPROVED** [review]

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 2]
```diff
@@ -995,6 +995,28 @@ public class CustomLoadBalancerConfiguration {
 
 WARNING: `HealthCheckServiceInstanceListSupplier` has its own caching mechanism based on Reactor Flux `replay()`. Therefore, if it's being used, you may want to skip wrapping that supplier with `CachingServiceInstanceListSupplier`.
 
+=== Same instance preference for LoadBalancer
+
+It is possible to set up the LoadBalancer in such a way that it will prefer the instance that was previously selected, if that instance is available.
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:1000]: Change to:

You can set up the `LoadBalancer` in such a way that it prefers the instance that was previously selected, if that instance is available.

File: docs/src/main/asciidoc/spring-cloud-commons.adoc [hunk 3]
```diff
@@ -995,6 +995,28 @@ public class CustomLoadBalancerConfiguration {
...
+
+For that, you will need to use the `SameInstancePreferenceServiceInstanceListSupplier`. It can be configured either by setting the value of `spring.cloud.loadbalancer.configurations` to `same-instance-preference` or by providing your own `ServiceInstanceListSupplier` bean, for example:
```
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:1002]: Change to:

For that, you need to use `SameInstancePreferenceServiceInstanceListSupplier`. You can configure it either by setting the value of `spring.cloud.loadbalancer.configurations` to `same-instance-preference` or by providing your own `ServiceInstanceListSupplier` bean -- for example:

File: docs/src/main/asciidoc/spring-cloud-commons.adoc (hunk 1 above)
Comment (Buzzardo) [inline docs/src/main/asciidoc/spring-cloud-commons.adoc:1019]: Add backticks around StickyRule.

Comment (Buzzardo) [reply docs/src/main/asciidoc/spring-cloud-commons.adoc:1019]: Definitely. Please add links between the two documents as well. Thanks.

**Review by Buzzardo — CHANGES_REQUESTED** [review]

**Comment by spencergibb** [comment]
This is global per load-balancer correct? For example in gateway, this would reuse the same instance across requests.

**Pull request merged**